src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceConnection.java

```diff
@@ -36,17 +36,18 @@
     private String serviceRootUrl;
     private Token token;
     private boolean enableArrowStream = false;
+    private boolean isPrestoPaginatedRequest = true;
 
     public QueryServiceConnection(String url, Properties properties) {
         this.properties = properties;
         this.serviceRootUrl = getServiceRootUrl(url);
         this.properties.put(Constants.LOGIN_URL, serviceRootUrl);
         setupDefaultClientSecretsIfRequired(serviceRootUrl, this.properties);
-        if(this.properties.containsKey(Constants.ENABLE_ARROW_STREAM)) {
-            if(this.properties.get(Constants.ENABLE_ARROW_STREAM).equals("true")) {
-                enableArrowStream = true;
-            }
-        }
+
+        // default `enableArrowStream` is false
+        enableArrowStream = Constants.TRUE_STR.equalsIgnoreCase((String) this.properties.get(Constants.ENABLE_ARROW_STREAM));
+        // default `isPrestoPaginatedRequest` is true
```

Comment:
won't properties.get throw exception if the key is not found ?

Reply:
equalsIgnoreCase handles null, so this would be safe. I have tested this at my local.